core/src/main/java/org/apache/iceberg/PartitionsTable.java

```diff
@@ -45,9 +45,10 @@
     super(ops, table, name);
 
     this.schema = new Schema(
-        Types.NestedField.required(1, "partition", table.spec().partitionType()),
-        Types.NestedField.required(2, "record_count", Types.LongType.get()),
-        Types.NestedField.required(3, "file_count", Types.IntegerType.get())
+        Types.NestedField.required(1, "spec_id", Types.IntegerType.get()),
```

Comment:
It is fine to add a new field, but you should not change the IDs of the other fields.

Reply:
Okay, in this case spec_id will be appended to the schema then.